### ISSUE TITLE
Fix S1172 FP: Triggers false positive on expressions with new() { ... } syntax

### ISSUE THREAD

**Issue by AHarman** [SonarSource/sonar-dotnet#7498]
### Description

S1172 triggers a false positive - parameters are used but a warning is applied anyway. This is appearing in Visual Studio, but not SonarCloud.

### Repro steps
This triggers it: 
```c#
private static MyClass CreateObj(string param1, string param2) => new()
{
    Prop1 = param1,
    Prop2 = param2,
    Prop3 = "a value"
};
```

### Expected behavior

The warning does not appear, because the parameters are used.

### Actual behavior

The warning appears, even though the parameters are used

### Known workarounds

The following examples do not trigger the false positive:

```c#
private static MyClass CreateObj(string param1, string param2) => new MyClass()
{
    Prop1 = param1,
    Prop2 = param2,
    Prop3 = "a value"
};
```

```c#
private static MyClass CreateObj(string param1, string param2) {
    return new()
    {
        Prop1 = param1,
        Prop2 = param2,
        Prop3 = "a value"
    };
}
```

### Related information

* C#/VB.NET Plugins version
  * C# 11
* Visual Studio version
  * 17.4.4
* MSBuild / dotnet version
  * .Net 7
* SonarScanner for .NET version (if used)
* Operating System
  * Windows 10

Not mentioned in the default list, but I'm using SonarLint for Visual Studio - 6.16.0.69538

**Comment by zsolt-kolbay-sonarsource** [comment]
Hi @AHarman.
I've been unable to replicate this false positive in Visual Studio 17.6 with SonarLint 6.16.0.69538.
Could you check if the problem still appears if you upgrade Visual Studio from 17.4.4 to the latest release?
If it does then please attach the zipped sample project here.

**Comment by zsolt-kolbay-sonarsource** [comment]
Hi @AHarman. Thank you for the attached project. The issue is confirmed as False Positive.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1724](https://sonarsource.atlassian.net/browse/NET-1724)

[NET-1724]: https://sonarsource.atlassian.net/browse/NET-1724?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
